packages/apollo-server-errors/src/index.ts

```diff
@@ -134,12 +134,16 @@ export function toApolloError(
 
 export interface ErrorOptions {
   code?: string;
+  // Sometimes it's more convenient to pass a function than a class name.
+  errorConstructor?: (message: string) => ApolloError;
   errorClass?: typeof ApolloError;
 }
 
 export function fromGraphQLError(error: GraphQLError, options?: ErrorOptions) {
   const copy: ApolloError =
-    options && options.errorClass
+    options && options.errorConstructor
+      ? options.errorConstructor(error.message)
```

Comment:
I did this because it turns out that `UserInputError` is not `typeof ApolloError`. The `ApolloError` constructor is `(message: string, code?: string, extensions?: Record<string, any>)` whereas the `UserInputError` constructor is `(message: string, properties?: Record<string, any>)` so there are valid calls to `new ApolloError("x", "y")` that can't be invoked on `new UserInputError`. (Other subclasses in this file just have `(message: string)` and I guess that's OK because additional arguments are just ignored?)

Now, fromGraphQLError only ever passes the first argument so both of those should be fine. But it doesn't work with `typeof ApolloError`. I did ask on [internal Slack](https://apollograph.slack.com/archives/C8MLN9S9H/p1617753079038400) to see if anyone has any ideas.

Reply:
Ah OK I think I figured it out.